table/block.cc

```diff
@@ -223,6 +224,120 @@ void DataBlockIter::Seek(const Slice& target) {
   }
 }
 
+// Optimized Seek for point lookup for an internal key `target`
+// target = "seek_user_key @ seqno".
+//
+// If the return value is FALSE, iter location is undefined, and it means:
+// 1) there is no key in this block falling into the range:
+//    ["seek_user_key @ seqno", "seek_user_key @ 0"], inclusive; AND
+// 2) the last key of this block has a greater user_key from seek_user_key
+//
+// If the return value is TRUE, iter location has two possibilies:
+// 1) If iter is valid, it is set to a location as if set by BinarySeek. In
+//    this case, it points to the first key_ with a larger user_key or a
+//    matching user_key with a seqno no greater than the seeking seqno.
+// 2) If the iter is invalid, it means either the block has no such user_key,
+//    or the block ends with a matching user_key but with a larger seqno.
+bool DataBlockIter::SeekForGet(const Slice& target) {
+  if (!data_block_hash_index_) {
+    Seek(target);
+    return true;
+  }
```

Comment:
Can we move this
```
  if (!data_block_hash_index_) {
    Seek(target);
    return true;
  }
```
to an inlined function, and leave the other part to a fallback function? I'm concerned that some minor performance regression will be caused in the non-hash use case, because the whole function cannot be inlined.